sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/tests/ScaleHostEndToEndTests.cs

```diff
@@ -0,0 +1,257 @@
+
+namespace Microsoft.Azure.WebJobs.Extensions.ServiceBus.Tests
+{
+    internal class ScaleHostEndToEndTests : WebJobsServiceBusTestBase
+    {
+        private const string Function1Name = "Function1";
+        private const string Function2Name = "Function2";
+
+        public ScaleHostEndToEndTests() : base(isSession: false)
+        {
+        }
+
+        [Test]
+        [TestCase(false)]
+        [TestCase(true)]
+        public async Task ScaleHostEndToEndTest(bool tbsEnabled)
+        {
+            string hostJson =
+            @"{
+                ""azureWebJobs"" : {
+                    ""extensions"": {
+                        ""serviceBus"": {
+                            ""maxConcurrentCalls"" :  1
+                        }
+                    }
+                }   
+            }";
+
+            // Function1Name uses connection string
+            // Function2Name uses AzureComponentFactory - simulating managed identity scenario in ScaleController
+            string triggers = $@"{{
+""triggers"": [
+    {{
+        ""name"": ""myQueueItem"",
+        ""type"": ""serviceBusTrigger"",
+        ""direction"": ""in"",
+        ""queueName"": ""{FirstQueueScope.QueueName}"",
+        ""functionName"": ""{Function1Name}""
+    }},
+    {{
+        ""name"": ""myQueueItem"",
+        ""type"": ""serviceBusTrigger"",
+        ""direction"": ""in"",
+        ""queueName"": ""{SecondQueueScope.QueueName}"",
+        ""connection"": ""ServiceBusConnection2"",
+        ""functionName"": ""{Function2Name}""
+    }}
+ ]}}";
+
+            IHost host = new HostBuilder().ConfigureServices(services => services.AddAzureClientsCore()).Build();
+            AzureComponentFactory defaultAzureComponentFactory = host.Services.GetService<AzureComponentFactory>();
+            AzureComponentFactoryWrapper factoryWrapper = new AzureComponentFactoryWrapper(defaultAzureComponentFactory, ServiceBusTestEnvironment.Instance.Credential);
+
+            string hostId = "test-host";
+            var loggerProvider = new TestLoggerProvider();
+
+            IHostBuilder hostBuilder = new HostBuilder();
+            hostBuilder.ConfigureLogging(configure =>
+            {
+                configure.SetMinimumLevel(LogLevel.Debug);
+                configure.AddProvider(loggerProvider);
+            });
+            hostBuilder.ConfigureAppConfiguration((hostBuilderContext, config) =>
+            {
+                // Adding host.json here
+                config.AddJsonStream(new MemoryStream(Encoding.UTF8.GetBytes(hostJson)));
+
+                var settings = new Dictionary<string, string>();
+                settings.Add("AzureWebJobsServiceBus", ServiceBusTestEnvironment.Instance.ServiceBusConnectionString);
+                settings.Add("ServiceBusConnection2:fullyQualifiedNamespace", $"{SecondQueueScope.NamespaceName}.servicebus.windows.net");
+                settings.Add("HostingConfig:Microsoft.Azure.WebJobs.Extensions.ServiceBus", "1");
```

Comment:
App settings is probably not the best place to park the hosting config, as the config is at the stamp level (not app)

Reply:
The settings are set for a scale host (one scale host per function app) so it's ok have then in `IConfiguration`